libvast/src/type.cpp

```diff
@@ -1113,6 +1113,39 @@ detail::generator<type::attribute_view> type::attributes() const& noexcept {
   __builtin_unreachable();
 }
 
+detail::generator<type> type::aliases() const noexcept {
+  const auto* root = &table(transparent::no);
+  while (true) {
+    switch (root->type_type()) {
+      case fbs::type::Type::NONE:
+      case fbs::type::Type::bool_type:
+      case fbs::type::Type::integer_type:
+      case fbs::type::Type::count_type:
+      case fbs::type::Type::real_type:
+      case fbs::type::Type::duration_type:
+      case fbs::type::Type::time_type:
+      case fbs::type::Type::string_type:
+      case fbs::type::Type::pattern_type:
+      case fbs::type::Type::address_type:
+      case fbs::type::Type::subnet_type:
+      case fbs::type::Type::enumeration_type:
+      case fbs::type::Type::list_type:
+      case fbs::type::Type::map_type:
+      case fbs::type::Type::record_type:
+        co_return;
+      case fbs::type::Type::enriched_type: {
+        const auto* enriched_type = root->type_as_enriched_type();
+        if (enriched_type->name())
```

Comment:
Is this check an internal "hack" to differentiate against anonymous types with attributes?

Reply:
`name()` here does not return the string view—instead, it returns a `flatbuffers::Indirect<T>*`, which holds a `nullptr` if the field is unset. So this is not a hack, but rather the proper way to check whether the field is set.